### PR TITLE
Ajax: remove extra "?" in the URL

### LINKED ISSUE #1791: $.getJSON treats request as JSONP when URL contains ?? (two question marks)
Originally reported by Stijn at: http://bugs.jquery.com/ticket/15216
<p>
Similar to <a href="http://bugs.jquery.com/ticket/8417">http://bugs.jquery.com/ticket/8417</a> and its duplicates, but they're all about the data parameter, not the URL parameter.
</p>
<p>
Happens with current stable and with edge. Fiddle: <a class="ext-link" href="http://jsfiddle.net/r5zcw6wx/" rel="nofollow"><span class="icon">&nbsp;</span>http://jsfiddle.net/r5zcw6wx/</a>
</p>
<p>
Code:
</p>
<pre class="wiki">$.getJSON("http://date.jsontest.com/")
$.getJSON("http://date.jsontest.com/?foo")
$.getJSON("http://date.jsontest.com/??foo")
</pre><p>
Requests:
</p>
<pre class="wiki">GET http://date.jsontest.com/
GET http://date.jsontest.com/?foo
GET http://date.jsontest.com/jQuery2129957077760988448071399739372034175163514043816_1408610547781foo?_=1408610547782
</pre><p>
Of course we shouldn't be making requests with two question marks (naive string concat on our side), and this could be a dupe of the mentioned ticket, but I figured this was worth reporting anyway.
</p><p>Issue reported for jQuery 2.1.1</p>

### PR DESCRIPTION
**Pull request by Mr21**
Hi :)

Fixes #1791

### CODE REVIEW

**Comment by mzgol** [comment]
@Mr21 What's the size difference?

File: src/ajax.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ define([
 
 var
 	rhash = /#.*$/,
+	rextraQuestions = /\?+/,
```
Comment (mzgol) [inline src/ajax.js]: Perhaps `/\?\?+/` would be faster as in this PR you're currently changing single question marks to themselves.

Comment (Mr21) [inline src/ajax.js]: Yeah right, but it's not a problem if this url change?
`"/file.html?q=??"`

Comment (mzgol) [inline src/ajax.js]: Ah, I see, you're changing the first question mark only and that's why it needs to be one. Seems fine as it is then.

Comment (jaubourg) [inline src/ajax.js]: Actually it isn't. The `??` replacer was introduced so that you c

Comment (Mr21) [inline src/ajax.js]: @jaubourg: i never do some `jsonp` but it's not only these cases:
`index.html?callback=?`
`index.html?var=1&callback=?`
`http://ex.com/?callback=?` and dont find where we have to write two `?`.

**Comment by Mr21** [comment]
@mzgol 
   raw     gz Compared to master @ 9d820fbde6d89bc7a06e2704be61cf6c0b4d6e3c
  +109    +40 dist/jquery.js
   +25     +8 dist/jquery.min.js

File: src/ajax.js (hunk 1 above)
Comment (jaubourg) [inline src/ajax.js]: http://bugs.jquery.com/ticket/4897

Comment (Mr21) [inline src/ajax.js]: @jaubourg oh.. ok.
I read these tests: https://github.com/jquery/jquery/blob/master/test/unit/ajax.js#L704
Finally, I don't understand exactly what is it necessary for doing some `jsonp`...
The doc http://api.jquery.com/jquery.ajax/ doesn't speak about `??` anywhere :(

Comment (jaubourg) [inline src/ajax.js]: Not the first time we forgot to document something ;)

Not sure I understand your question but the fix (ie introduction of `??`) was for APIs that would have the callback as part of the URL, not as a parameter (see [the original ticket](http://bugs.jquery.com/ticket/4897)).

Comment (Mr21) [inline src/ajax.js]: Ahhhh I understood :)

It's really different with that behaviour :/
Because if we want to be friendly on `.html??var=` we have to be also friendly on `/callback=???var=` and `/callback=?/??var=` the regex is more complex and maybe it's not so necessary.

I think we have to use this regex: `.replace( /([^=]|=\?)\?+/, "$1?" )`
(some tests here: http://jsfiddle.net/Mr21/0sofoaay/)

Comment (jaubourg) [inline src/ajax.js]: Well, to be honest, I think we shouldn't try and fix anything because it is working as intended. The original bug report is bogus in that it is asking to cater to an illformed url.

**Comment by mzgol** [comment]
@timmywil why did you remove the "Needs review" label? Shouldn't we discuss it?

**Comment by dmethvin** [comment]
My take is [similar to @jaubourg](https://github.com/jquery/jquery/pull/2490/files#r37265083), this is a malformed URL because the second `?` should be URL encoded. For better or worse we are using `??` to have special meaning so it seems strange to subvert that in order to avoid mangling an already-invalid URL.

**Comment by markelog** [comment]
@Mr21 we decided that this is notabug, nevertheless, thank you for the effort!

**Comment by Mr21** [comment]
Right, no problem :) what's jaubourg told me increased a lot the complexity of the regex anyway...
